File: app/src/main/java/com/androidevlinux/percy/UTXO/fragments/BaseFragment.java
package com.androidevlinux.percy.UTXO.fragments;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.androidevlinux.percy.UTXO.data.network.ApiManager;

public class BaseFragment extends Fragment {
    protected ApiManager apiManager;
    protected Activity mActivity;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mActivity = (Activity) context;
    }

    @Override
    public void onDetach() {
        super.onDetach();
        this.mActivity = null;
    }

    @Override
    public void onStop() {
        super.onStop();
    }


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        apiManager = ApiManager.getInstance();
    }
}

File: app/src/main/java/com/androidevlinux/percy/UTXO/fragments/BaseFragment.kt
package com.androidevlinux.percy.UTXO.fragments

import android.app.Activity
import android.content.Context
import android.os.Bundle
import com.androidevlinux.percy.UTXO.data.network.ApiManager

open class BaseFragment : androidx.fragment.app.Fragment() {
    protected var apiManager: ApiManager? = null
    protected var mActivity: Activity? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mActivity = context as Activity?
    }

    override fun onDetach() {
        super.onDetach()
        this.mActivity = null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        apiManager = ApiManager.instance
    }
}
